README.md

```diff
@@ -0,0 +1,105 @@
+                    "Pattern": "(?i)(ESR)",
+                    "Template": "[ESR](https://docs.mattermost.com/process/training.html#esr)"
+                },
+                {
+                    "Pattern": "((?P<level>0|1|2|3|4|5)/5)",
+                    "Template": "[${level}/5](https://docs.mattermost.com/process/training.html#id8)"
+                },
+                {
+                    "Pattern": "(MM)(-)(?P<jira_id>\\d+)",
+                    "Template": "[MM-${jira_id}](https://mattermost.atlassian.net/browse/MM-${jira_id})"
+                },
+                {
+                    "Pattern": "https://pre-release\\.mattermost\\.com/core/pl/(?P<id>[a-zA-Z0-9]+)",
+                    "Template": "[<jump to convo>](https://pre-release.mattermost.com/core/pl/${id})"
+                },
+                {
+                    "Pattern": "(https://mattermost\\.atlassian\\.net/browse/)(MM)(-)(?P<jira_id>\\d+)",
+                    "Template": "[MM-${jira_id}](https://mattermost.atlassian.net/browse/MM-${jira_id})"
+                },
+                {
+                    "Pattern": "https://github\\.com/mattermost/(?P<repo>.+)/pull/(?P<id>\\d+)",
+                    "Template": "[pr-${repo}-${id}](https://github.com/mattermost/${repo}/pull/${id})"
+                },
+                {
+                    "Pattern": "https://github\\.com/mattermost/(?P<repo>.+)/issues/(?P<id>\\d+)",
+                    "Template": "[issue-${repo}-${id}](https://github.com/mattermost/${repo}/issues/${id})"
+                },
+                {
+                    "Pattern": "(PLT)(-)(?P<jira_id>\\d+)",
+                    "Template": "[PLT-${jira_id}](https://mattermost.atlassian.net/browse/PLT-${jira_id})"
+                },
+                {
+                    "Pattern": "(https://mattermost\\.atlassian\\.net/browse/)(PLT)(-)(?P<jira_id>\\d+)",
+                    "Template": "[PLT-${jira_id}](https://mattermost.atlassian.net/browse/PLT-${jira_id})"
+                }
+            ]
+        },
+    },
+    ...
+    "PluginStates": {
+        ...
+        "mattermost-autolink": {
+            "Enable": true
+        },
+        ...
+    }
+},
+```
+
+## Examples
+
+1. Autolinking `Ticket ####:text with alphanumberic characters and spaces` to a ticket link. Use:
+  - Pattern: `(?i)(ticket )(?P<ticket_id>.+)(:)(?P<ticket_info>.*)`, or if the ticket_id is a number, then `(?i)(ticket )(?P<ticket_id>\d+)(:)(?P<ticket_info>.*)`
+  - Template: `[Ticket ${ticket_id}: ${ticket_info}](https://github.com/mattermost/mattermost-server/issues/${ticket_id})`
+
+2. Autolinking a link to a GitHub PR to a link with format "pr-repo-id". Use:
+  - Pattern: `https://github\\.com/mattermost/(?P<repo>.+)/pull/(?P<id>\\d+)`
+  - Template: `[pr-${repo}-${id}](https://github.com/mattermost/${repo}/pull/${id})`
+
+3. Using autolinking to create group mentions. Use:
+  - Pattern: `@customgroup*`
+  - Template: `[@customgroup]( \\* @user1 @user2 @user3 \\* )`
```

Comment:
Add some examples which make use of DisableNonWordPrefix and DisableNonWordSuffix

Reply:
Add documentation to README for the following:
- DisableNonWordPrefix
- DisableNonWordSuffix
- Scope